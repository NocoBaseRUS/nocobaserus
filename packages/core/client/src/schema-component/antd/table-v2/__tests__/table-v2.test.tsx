import React from 'react';
import { render, sleep } from 'testUtils';
import App1 from '../demos/demo1';
import App2 from '../demos/demo2';

describe('TableV2', () => {
  it('basic', async () => {
    const { container } = render(<App1 />);

    await sleep(300);
    expect(container).toMatchSnapshot();
  });

  it('tree table', () => {
    const { container } = render(<App2 />);
    expect(container).toMatchSnapshot();
  });
});
