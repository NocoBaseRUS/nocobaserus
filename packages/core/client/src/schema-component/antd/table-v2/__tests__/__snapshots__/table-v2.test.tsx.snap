// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TableV2 > basic 1`] = `
<div>
  <div>
    <div
      class="nb-fixed-block"
    >
      <button
        class="ant-btn ant-btn-default css-49hvkh"
        type="button"
      >
        <span>
          刷新
        </span>
      </button>
      <div
        class="css-1s4dz4u"
      >
        <div
          class="ant-table-wrapper"
        >
          <div
            class="ant-spin-nested-loading"
          >
            <div
              class="ant-spin-container"
            >
              <div
                class="ant-table ant-table-scroll-horizontal"
              >
                <div
                  class="ant-table-container"
                >
                  <div
                    class="ant-table-content"
                    style="overflow-x: auto; overflow-y: hidden;"
                  >
                    <table
                      style="min-width: 100%; table-layout: auto;"
                    >
                      <colgroup>
                        <col
                          class="ant-table-selection-col"
                        />
                        <col
                          style="width: 200px;"
                        />
                        <col
                          style="width: 200px;"
                        />
                      </colgroup>
                      <thead
                        class="ant-table-thead"
                      >
                        <tr>
                          <th
                            class="ant-table-cell ant-table-selection-column css-p6py3w"
                          >
                            <div
                              class="ant-table-selection"
                            >
                              <label
                                class="ant-checkbox-wrapper"
                              >
                                <span
                                  class="ant-checkbox"
                                >
                                  <input
                                    aria-label="Select all"
                                    class="ant-checkbox-input"
                                    type="checkbox"
                                    value=""
                                  />
                                  <span
                                    class="ant-checkbox-inner"
                                  />
                                </span>
                              </label>
                            </div>
                          </th>
                          <th
                            class="ant-table-cell css-p6py3w"
                          >
                            <div>
                              Name
                            </div>
                          </th>
                          <th
                            class="ant-table-cell css-p6py3w"
                          >
                            <div>
                              Actions
                            </div>
                          </th>
                        </tr>
                      </thead>
                      <tbody
                        class="ant-table-tbody"
                      >
                        <tr
                          aria-hidden="true"
                          class="ant-table-measure-row"
                          style="height: 0px; font-size: 0px;"
                        >
                          <td
                            style="padding: 0px; border: 0px; height: 0px;"
                          >
                            <div
                              style="height: 0px; overflow: hidden;"
                            >
                               
                            </div>
                          </td>
                          <td
                            style="padding: 0px; border: 0px; height: 0px;"
                          >
                            <div
                              style="height: 0px; overflow: hidden;"
                            >
                               
                            </div>
                          </td>
                          <td
                            style="padding: 0px; border: 0px; height: 0px;"
                          >
                            <div
                              style="height: 0px; overflow: hidden;"
                            >
                               
                            </div>
                          </td>
                        </tr>
                        <tr
                          class="ant-table-row ant-table-row-level-0"
                          data-row-key="1"
                        >
                          <td
                            class="ant-table-cell ant-table-selection-column css-1omg1s0"
                          >
                            <div
                              class="css-xl522n"
                            >
                              <div
                                class="css-2mnx34"
                              >
                                <div
                                  class="nb-table-index"
                                  style="padding: 0px 8px 0px 16px;"
                                >
                                  1
                                </div>
                              </div>
                              <div
                                class="nb-origin-node css-cdm7wh"
                              >
                                <label
                                  class="ant-checkbox-wrapper"
                                >
                                  <span
                                    class="ant-checkbox"
                                  >
                                    <input
                                      class="ant-checkbox-input"
                                      type="checkbox"
                                      value=""
                                    />
                                    <span
                                      class="ant-checkbox-inner"
                                    />
                                  </span>
                                </label>
                              </div>
                            </div>
                          </td>
                          <td
                            class="ant-table-cell css-1omg1s0"
                          >
                            <div
                              class="ant-description-input"
                            >
                              name1
                            </div>
                          </td>
                          <td
                            class="ant-table-cell css-1omg1s0"
                          >
                            <button
                              class="ant-btn ant-btn-default css-49hvkh"
                              type="button"
                            >
                              <span>
                                View
                              </span>
                            </button>
                          </td>
                        </tr>
                        <tr
                          class="ant-table-row ant-table-row-level-0"
                          data-row-key="2"
                        >
                          <td
                            class="ant-table-cell ant-table-selection-column css-1omg1s0"
                          >
                            <div
                              class="css-xl522n"
                            >
                              <div
                                class="css-2mnx34"
                              >
                                <div
                                  class="nb-table-index"
                                  style="padding: 0px 8px 0px 16px;"
                                >
                                  2
                                </div>
                              </div>
                              <div
                                class="nb-origin-node css-cdm7wh"
                              >
                                <label
                                  class="ant-checkbox-wrapper"
                                >
                                  <span
                                    class="ant-checkbox"
                                  >
                                    <input
                                      class="ant-checkbox-input"
                                      type="checkbox"
                                      value=""
                                    />
                                    <span
                                      class="ant-checkbox-inner"
                                    />
                                  </span>
                                </label>
                              </div>
                            </div>
                          </td>
                          <td
                            class="ant-table-cell css-1omg1s0"
                          >
                            <div
                              class="ant-description-input"
                            >
                              name2
                            </div>
                          </td>
                          <td
                            class="ant-table-cell css-1omg1s0"
                          >
                            <button
                              class="ant-btn ant-btn-default css-49hvkh"
                              type="button"
                            >
                              <span>
                                View
                              </span>
                            </button>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
              <ul
                class="ant-pagination ant-table-pagination ant-table-pagination-right"
              >
                <li
                  class="ant-pagination-total-text"
                >
                  Total 100 items
                </li>
                <li
                  aria-disabled="true"
                  class="ant-pagination-prev ant-pagination-disabled"
                  title="Previous Page"
                >
                  <button
                    class="ant-pagination-item-link"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      aria-label="left"
                      class="anticon anticon-left"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="left"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                        />
                      </svg>
                    </span>
                  </button>
                </li>
                <li
                  class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
                  tabindex="0"
                  title="1"
                >
                  <a
                    rel="nofollow"
                  >
                    1
                  </a>
                </li>
                <li
                  class="ant-pagination-item ant-pagination-item-2"
                  tabindex="0"
                  title="2"
                >
                  <a
                    rel="nofollow"
                  >
                    2
                  </a>
                </li>
                <li
                  class="ant-pagination-item ant-pagination-item-3"
                  tabindex="0"
                  title="3"
                >
                  <a
                    rel="nofollow"
                  >
                    3
                  </a>
                </li>
                <li
                  class="ant-pagination-item ant-pagination-item-4"
                  tabindex="0"
                  title="4"
                >
                  <a
                    rel="nofollow"
                  >
                    4
                  </a>
                </li>
                <li
                  class="ant-pagination-item ant-pagination-item-5 ant-pagination-item-before-jump-next"
                  tabindex="0"
                  title="5"
                >
                  <a
                    rel="nofollow"
                  >
                    5
                  </a>
                </li>
                <li
                  class="ant-pagination-jump-next ant-pagination-jump-next-custom-icon"
                  tabindex="0"
                  title="Next 5 Pages"
                >
                  <a
                    class="ant-pagination-item-link"
                  >
                    <div
                      class="ant-pagination-item-container"
                    >
                      <span
                        aria-label="double-right"
                        class="anticon anticon-double-right ant-pagination-item-link-icon"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="double-right"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M533.2 492.3L277.9 166.1c-3-3.9-7.7-6.1-12.6-6.1H188c-6.7 0-10.4 7.7-6.3 12.9L447.1 512 181.7 851.1A7.98 7.98 0 00188 864h77.3c4.9 0 9.6-2.3 12.6-6.1l255.3-326.1c9.1-11.7 9.1-27.9 0-39.5zm304 0L581.9 166.1c-3-3.9-7.7-6.1-12.6-6.1H492c-6.7 0-10.4 7.7-6.3 12.9L751.1 512 485.7 851.1A7.98 7.98 0 00492 864h77.3c4.9 0 9.6-2.3 12.6-6.1l255.3-326.1c9.1-11.7 9.1-27.9 0-39.5z"
                          />
                        </svg>
                      </span>
                      <span
                        class="ant-pagination-item-ellipsis"
                      >
                        •••
                      </span>
                    </div>
                  </a>
                </li>
                <li
                  class="ant-pagination-item ant-pagination-item-50"
                  tabindex="0"
                  title="50"
                >
                  <a
                    rel="nofollow"
                  >
                    50
                  </a>
                </li>
                <li
                  aria-disabled="false"
                  class="ant-pagination-next"
                  tabindex="0"
                  title="Next Page"
                >
                  <button
                    class="ant-pagination-item-link"
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      aria-label="right"
                      class="anticon anticon-right"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="right"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                        />
                      </svg>
                    </span>
                  </button>
                </li>
                <li
                  class="ant-pagination-options"
                >
                  <div
                    aria-label="Page Size"
                    class="ant-select ant-pagination-options-size-changer ant-select-single ant-select-show-arrow"
                  >
                    <div
                      class="ant-select-selector"
                    >
                      <span
                        class="ant-select-selection-search"
                      >
                        <input
                          aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                          aria-autocomplete="list"
                          aria-controls="rc_select_TEST_OR_SSR_list"
                          aria-expanded="false"
                          aria-haspopup="listbox"
                          aria-label="Page Size"
                          aria-owns="rc_select_TEST_OR_SSR_list"
                          autocomplete="off"
                          class="ant-select-selection-search-input"
                          id="rc_select_TEST_OR_SSR"
                          readonly=""
                          role="combobox"
                          style="opacity: 0;"
                          type="search"
                          unselectable="on"
                          value=""
                        />
                      </span>
                      <span
                        class="ant-select-selection-item"
                        title="2 / page"
                      >
                        2 / page
                      </span>
                    </div>
                    <span
                      aria-hidden="true"
                      class="ant-select-arrow"
                      style="user-select: none;"
                      unselectable="on"
                    >
                      <span
                        aria-label="down"
                        class="anticon anticon-down ant-select-suffix"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="down"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                          />
                        </svg>
                      </span>
                    </span>
                  </div>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`TableV2 > tree table 1`] = `
<div>
  <div
    class="ant-table-wrapper"
  >
    <div
      class="ant-spin-nested-loading"
    >
      <div
        class="ant-spin-container"
      >
        <div
          class="ant-table"
        >
          <div
            class="ant-table-container"
          >
            <div
              class="ant-table-content"
            >
              <table
                style="table-layout: auto;"
              >
                <colgroup>
                  <col
                    class="ant-table-selection-col"
                  />
                </colgroup>
                <thead
                  class="ant-table-thead"
                >
                  <tr>
                    <th
                      class="ant-table-cell ant-table-selection-column"
                    >
                      <div
                        class="ant-table-selection"
                      >
                        <label
                          class="ant-checkbox-wrapper"
                        >
                          <span
                            class="ant-checkbox"
                          >
                            <input
                              aria-label="Select all"
                              class="ant-checkbox-input"
                              type="checkbox"
                              value=""
                            />
                            <span
                              class="ant-checkbox-inner"
                            />
                          </span>
                        </label>
                      </div>
                    </th>
                    <th
                      class="ant-table-cell"
                    />
                  </tr>
                </thead>
                <tbody
                  class="ant-table-tbody"
                >
                  <tr
                    class="ant-table-row ant-table-row-level-0"
                    data-row-key="1"
                  >
                    <td
                      class="ant-table-cell ant-table-selection-column"
                    >
                      <label
                        class="ant-checkbox-wrapper"
                      >
                        <span
                          class="ant-checkbox"
                        >
                          <input
                            class="ant-checkbox-input"
                            type="checkbox"
                            value=""
                          />
                          <span
                            class="ant-checkbox-inner"
                          />
                        </span>
                      </label>
                    </td>
                    <td
                      class="ant-table-cell ant-table-cell-with-append"
                    >
                      <span
                        class="ant-table-row-indent indent-level-0"
                        style="padding-left: 0px;"
                      />
                      <button
                        aria-expanded="false"
                        aria-label="Expand row"
                        class="ant-table-row-expand-icon ant-table-row-expand-icon-spaced"
                        type="button"
                      />
                      <li>
                        value: 
                        t1
                      </li>
                    </td>
                  </tr>
                  <tr
                    class="ant-table-row ant-table-row-level-0"
                    data-row-key="2"
                  >
                    <td
                      class="ant-table-cell ant-table-selection-column"
                    >
                      <label
                        class="ant-checkbox-wrapper"
                      >
                        <span
                          class="ant-checkbox"
                        >
                          <input
                            class="ant-checkbox-input"
                            type="checkbox"
                            value=""
                          />
                          <span
                            class="ant-checkbox-inner"
                          />
                        </span>
                      </label>
                    </td>
                    <td
                      class="ant-table-cell ant-table-cell-with-append"
                    >
                      <span
                        class="ant-table-row-indent indent-level-0"
                        style="padding-left: 0px;"
                      />
                      <button
                        aria-expanded="false"
                        aria-label="Expand row"
                        class="ant-table-row-expand-icon ant-table-row-expand-icon-collapsed"
                        type="button"
                      />
                      <li>
                        value: 
                        t2
                      </li>
                    </td>
                  </tr>
                  <tr
                    class="ant-table-row ant-table-row-level-0"
                    data-row-key="3"
                  >
                    <td
                      class="ant-table-cell ant-table-selection-column"
                    >
                      <label
                        class="ant-checkbox-wrapper"
                      >
                        <span
                          class="ant-checkbox"
                        >
                          <input
                            class="ant-checkbox-input"
                            type="checkbox"
                            value=""
                          />
                          <span
                            class="ant-checkbox-inner"
                          />
                        </span>
                      </label>
                    </td>
                    <td
                      class="ant-table-cell ant-table-cell-with-append"
                    >
                      <span
                        class="ant-table-row-indent indent-level-0"
                        style="padding-left: 0px;"
                      />
                      <button
                        aria-expanded="false"
                        aria-label="Expand row"
                        class="ant-table-row-expand-icon ant-table-row-expand-icon-collapsed"
                        type="button"
                      />
                      <li>
                        value: 
                        t3
                      </li>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <ul
          class="ant-pagination ant-table-pagination ant-table-pagination-right"
        >
          <li
            aria-disabled="true"
            class="ant-pagination-prev ant-pagination-disabled"
            title="Previous Page"
          >
            <button
              class="ant-pagination-item-link"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                aria-label="left"
                class="anticon anticon-left"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="left"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                  />
                </svg>
              </span>
            </button>
          </li>
          <li
            class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
            tabindex="0"
            title="1"
          >
            <a
              rel="nofollow"
            >
              1
            </a>
          </li>
          <li
            aria-disabled="true"
            class="ant-pagination-next ant-pagination-disabled"
            title="Next Page"
          >
            <button
              class="ant-pagination-item-link"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                aria-label="right"
                class="anticon anticon-right"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="right"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                  />
                </svg>
              </span>
            </button>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
